### ISSUE TITLE
Missed calls interface

### ISSUE THREAD

**Issue by ajsco** [BelledonneCommunications/linphone-desktop#384]
**OS:** xubuntu 20.04 LTS
**Linphone version:** Linphone-4.2.0-alpha.73+0ac9f08.AppImage

I'm loving Linphone as a softphone because it works great in Linux. However there is a missing feature, or maybe a bug, that is a real deal breaker :(

When someone calls me and I do not answer, there is no interface to see the number that called me. I've to go to the logs in order to figure out who it was :(

Since the logs have the information and the "call-history.db" has it too, can you please add a GUI to see the missed calls? Basically an view of the "call-history.db" would do the trick :)

Thanks

**Comment by glebaccon** [comment]
Same for me with Linphone-4.2.0-alpha.68+9f1ae82.AppImage

The "call-history.db" has information but the GUI no.

**Comment by julonexus** [comment]
From https://github.com/BelledonneCommunications/linphone-desktop/issues/377, I guess you are on the default profile?

**Comment by ajsco** [comment]
Sorry I do not understand what you mean.
I created the account using "USE A SIP ACOCUNT". I'm connected to my company VoIP server.

Where are the profiles you are mentioning? If I can do something about it, please provide me with further instructions.

Thanks

**Comment by julonexus** [comment]
When you start the application, you have a default account. Something like `sip:ajsco@42.42.42.42:5060`. It is used to start dealing with servers.
You can select your accounts at the top left. But if you are on your "use account", it should work as expected : you should get all missed logs in your history.

**Comment by ajsco** [comment]
This is what I see
[Someone calling me OK](https://www.dropbox.com/s/rqaewjwa6aa8uqq/1of2_calling-me.png)

[I did not answer](https://www.dropbox.com/s/4qdu86w4t7s8dn5/2of2_not-answered.png)
I can not find out using the GUI who called me :(

**Comment by julonexus** [comment]
What do you see when you click on your account?

**Comment by ajsco** [comment]
Sorry, would you be willing to see for you self using Zoom or Google Meet? I can share my screen with you.

**Comment by ajsco** [comment]
![image](https://user-images.githubusercontent.com/26432003/84138902-ec78e500-aa46-11ea-85db-1539bcf1adb4.png)

**Comment by julonexus** [comment]
But there is no more bubble at the top left(behind the popup). It disappeared when you clicked on the account selection?

**Comment by ajsco** [comment]
I closes Linphone and open it again to get rid of it

**Comment by ajsco** [comment]
If I click the "1" next to the chat icon it does nothing or opens the active account screen I've just pasted before.

If I click the "1" in the taskbar that replaces the linphone icon, it open the context menu
![image](https://user-images.githubusercontent.com/26432003/84139822-7aa19b00-aa48-11ea-92b0-c74941766903.png)

**Comment by julonexus** [comment]
Ok, Unseen Missed call are not stored by the SDK. It's why there are no bubbles when you restart your app but logs should be shown when clicking on the contact. Could you confirm it?

I would like to know if the bubbles of missed calls are showing while your app is running because it was the goal of the previous fix.

**Comment by ajsco** [comment]
Hi!

Paula called me
![image](https://user-images.githubusercontent.com/26432003/84147008-901cc200-aa54-11ea-8276-ec88cc7fde8b.png)

and I cannot find any information clicking the CALLS button

![image](https://user-images.githubusercontent.com/26432003/84146928-6499d780-aa54-11ea-9f84-86594cf37460.png)

![image](https://user-images.githubusercontent.com/26432003/84146949-6f546c80-aa54-11ea-8690-46bcf9972bec.png)

**Comment by julonexus** [comment]
If you get the account selection, you should see a bubble.
But the strange thing is : the ";re..." at the end of the notification. I think this can lead to your issue that I could reproduce.

**Comment by ajsco** [comment]
Here a sample from the log

`2020-06-09 11:53:23:400 [AppRun.wrapped/belle-sip] MESSAGE channel [0x2c74d40]: message sent to [UDP://voip.iscte-iul.pt:5060], size: [554] bytes
SIP/2.0 487 Request terminated
Via: SIP/2.0/UDP 193.137.58.162:5060;branch=z9hG4bK-9861dc-5edf6a09-d9c94b4d-6c380bf1
From: "Paula Pinto" <sip:220329@voip.iscte-iul.pt;received-from=10.26.80.25:5060>;tag=a23a89c1-13c4-5edf6a09-d9c94b4d-50b39798
To: <sip:223013@voip.iscte-iul.pt>;tag=hDPEUOI
Call-ID: CXC-41-621fbe88-a23a89c1-13c4-5edf6a09-d9c94b4d-fc8910a@voip.iscte-iul.pt
CSeq: 1 INVITE
User-Agent: Linphone Desktop/4.2.0-alpha-73-g0ac9f08 (Ubuntu 20.04 LTS, Qt 5.12.5) LinphoneCore/4.4.0-beta-17-g49592e2
Supported: replaces, outbound, gruu`

The ";re" should be "received-from=10.26.80.25:5060", I think.

**Comment by julonexus** [comment]
Is Paula using Linphone? I don't think received-from is a standard. There is no occurences in RFC.
If she is using something free, I could test it for validation without hacking something

**Comment by glebaccon** [comment]
Sorry but I d'ont understand the section "Previously".

If I receive incoming call and I answer it : New entry in "Previously" list -> OK
If I receive incoming call and I d'ont answer it : No new entry -> Why ? It should be there
If I call someone and he answer me : New entry in "Previously" list -> OK
If I call someone and he d'ont answer me : No new entry -> Why ? It should be there

Also, if I restart the application the list is empty. Why ?

For me the list need to be filled for all situations. Only the GUI is affected, the 'call-history.db' is filled.

**Comment by julonexus** [comment]
@glebaccon `Previously` section shows you all contacts with which you had do something (chat/calls) with your active proxy(the one selected at the top left). Not having some is the current issue : The GUI cannot match what you have in your DB and your active proxy and I guess it's because of the URI that can be a mess.

@ajsco Could you share the databases folder? [Locations](https://wiki.linphone.org/xwiki/wiki/public/view/Linphone/Desktop%20application%20Files%20Tree/)
You can rename your linphone folder by linphonebak before launching the app. You will have a fresh database. And after making the buggy calls, you can send the call-history.db and linphone.db to sip:support_desktop@sip.linphone.org

**Comment by glebaccon** [comment]
in fact, some call status is not present in the history so I create a new issue #386

**Comment by ajsco** [comment]
@julonexus 

> @ajsco Could you share the databases folder? Locations
You can rename your linphone folder by linphonebak before launching the app. You will have a fresh database. And after making the buggy calls, it would be great if you could send the call-history.db and linphone.db to sip:support_desktop@sip.linphone.org

Sorry I was not able to provide you the information yet.
I will sent it until the end of the day.

Thanks

**Comment by ajsco** [comment]
@julonexus I've sent an email to "support_desktop@sip.linphone.org" and got the error message "

Address not found
--
Your message wasn't delivered to support_desktop@sip.linphone.org because the address couldn't be found, or is unable to receive mail.

"

**Comment by julonexus** [comment]
@ajsco I wrote 'sip:support_desktop@sip.linphone.org', it is a sip address. So you can send it with Linphone ;)
By the way, there were some fix about storing logs. Maybe they fixed your issue, you should try the alpha-92 or later

**Comment by ajsco** [comment]
Sorry for being a basic user regarding Linphone, but I still do not understand you can I send you the files you requested.
![image](https://user-images.githubusercontent.com/26432003/85375679-90c25780-b52e-11ea-988e-bab6bb9947b2.png)
I tried to use the chat feature, but I can not add an attachment. Can you please explain my step by step how can I send you the file?

**Comment by julonexus** [comment]
Ha ok. You have not configured your `file_transfer_server_url`. When using a server account that is not linphone, the field is not filled (see the Call&Chat tab in settings). So you have to fill it with your file transfert server url.

For a mail, you could send it to linphone-desktop@belledonne-communications.com with few details on the issue.

**Comment by julonexus** [comment]
@ajsco And have you still your issue with a linphone account?

**Comment by ajsco** [comment]
Hi @julonexus 

I've not tried using a Linphone account, because since people from office simply call the regular extension (for instance 223013) from the physical VoIP phone, I think if I use a Linphone account instead our own they will not be able to call me.

Finally I've send you the information you asked be email.

Thanks for your help and patience.

**Comment by julonexus** [comment]
@ajsco Thanks for your files. When testing, I could get all your missed calls from your DB so I couldn't reproduce.
I tested [alpha95](https://www.linphone.org/snapshots/AppImages/Linphone-4.2.0-alpha.95+3dd6b95b.AppImage)
Could you confirm me that you get all logs after restarting the app?(closing the window is not enough, you have to close it from the menu)

**Comment by ajsco** [comment]
Hi!
The missed calls do not appear in the "Previous" section on the left. Although they some seem to show in the search box, or maybe I'm just confused regarding the search box.

![image](https://user-images.githubusercontent.com/26432003/85743769-3d056900-b6fc-11ea-9984-488f485cb89d.png)

**Comment by ajsco** [comment]
@julonexus if the issue I'm reporting only happens to me, it might be a problem related to the VoIP software my company is using. I thought everyone was having this issue and needed the feature...

**Comment by julonexus** [comment]
@ajsco I just like spending my free time on resolving github issues ;)
For other issues where addresses are differents (like client and asterisk), it may be a configuration issue that is not from Linphone Application.
From your DB, I don't think yours is about a server issue as domains are correct. When reading databases, I can retrieve your missed call, but you cannot and I don't know why.

**Comment by dtheodoro** [comment]
Hi @julonexus . I'm experiencing the same problem has @ajsco 
I'm using 4.2.0 version with MacOS Catalina 10.15.5 (19F101)
I can see a ballon near my status on the upper left corner and also in the docker bar when I miss a incomming call.
This "chat" ballon is not clickable also.
I've just checked the call-history.db and it is on it with 'status' equal 2 but i can't see ir on the "Previously" window.

**Comment by dtheodoro** [comment]
Hi again.... now I figured out that all incoming call (not only missing) is not displayed on the "Previously" tab window.

**Comment by jason1987d** [comment]
Could this be similar to issue #397 ? I am having similar issues, I do see the titles of the contact for the thread under "previously", but all messages in there disappear, I have to check the logs for received messages.

**Comment by ernierasta** [comment]
Just want to confirm, that this is constant issue on linux (wayland), for almost year (I tried 3 times during last few months and issue was always there). When I decline or miss call - no info about that (and I have no idea who called to me). Last test - today - latest Linphone-4.2.2.AppImage - the same problem.

Reverting to 3.x solves that for me (and to be honest - I like gui much more).

But do not get me wrong, no complains, thank you for one of the best linux sip client!

**Comment by julonexus** [comment]
As I proposed in #397 , please check the INVITE message.
3.x may "solves" this issue, but the SDK (and the app) was too much permissive against RFC. So in this case, I'm afraid that you are stuck with the old version (unless your provider change its behaviour to not change the "To"as they want, that lead to differ from the REGISTER) . As for now, this is the only case that I met this missing call issue.

**Comment by ernierasta** [comment]
@julonexus thank you for response!

You are probably right, I see this:
```
INVITE sip:MY_USERNAME@my_pub_ip:5060
...
To: <sip:MY_PHONENAME@PROXY:5060>
```

The problem is, this is commercial SIP and probably there is not much I can do about (and it looks like - more providers has this problem). I understand why You want to promote RFC standards I completely agree with that.

But it doesn't change reality, that many of has to use some different client. Maybe some option in config, that allows for workaround this problem would be solution?

For me this problem is solved with 3.x, but others may prefer 4.x ...

**Comment by Ixreb** [comment]
First of all, thanks for the good work developing this software.

I have the same problem. I have received missed calls and I have received calls that I have answered, none of them appear.

My pbx domain is something like "pbx02.whatever.com" (external provider). It works perfect for outgoing calls, it's registered in the "Previously" section.

Also for internal calls it works fine. But when I receive external calls from real numbers I see something like "sip:69557XXXX@172.XX.3.31", and these calls are not registered in the "Previously" section (but I can see it in "call-history.db"). I see the icon with the number of missed calls that does not work, I only get the account selection.

The calls appear in the search bar, but if I try to call back it doesn't work, because it uses the "...@172.XX.3.31".

I have tried everything, and nothing works. Any ideas?

I can add captures or log if you need it.

**Comment by Ixreb** [comment]
I just saw that it also happens with the configuration of our other provider.

Attached images:
![image](https://i.ibb.co/rfG3HDx/linphone01.png)

![image](https://i.ibb.co/tpV7D2b/linphone02.png)

![image](https://i.ibb.co/TqbHtZ6/linphone03.png)

Here the account settings:
![image](https://i.ibb.co/pwrRWRW/linphone04.png)

**Comment by Hawkeye1909** [comment]
I'm facing the same issue. In call-history.db the callee column shows the default identiy although I selected my SIP account... If I call somebody the correct identity is set as caller. And if I overwrite the callee entry I can see the call in the history.

I'm on Windows and using version 4.2.2.

**Comment by tbelliard** [comment]
Same issue here with version Desktop 4.2.2 (Qt5.12.5 Core 4.4.0-13-gc99cb9c88) both on Linux and Windows. On Linux yesterday I had two missed calls and saw the notification bubble but no way to see the history, and on windows I just tested with a colleague by calling him (he did not answer) and he does not even get a notification.
SIP provider is OVH in France.

**Comment by julonexus** [comment]
The issue doesn't come from Linphone but from provider where the identity field doesn't match to the subscribed account.
A workaround has been done to show all call logs by clicking on the "previsouly" bar. It will be available in the next nightly from this [commit](8ecd1bc32c3f251b19b7b0f8425cb626aa1f5e15)
As long as you can get your incoming call, I'm closing this issue. Thanks for you feedbacks.
